### DISCUSSION TITLE
Problem in training spacy's ud_benchmark model

### DISCUSSION THREAD

**Discussion by megamattc** [explosion/spaCy#11912] [Help: Installation] [answered]
Hello,

I am trying to train spacy's ud_benchmark model (https://github.com/explosion/projects/tree/v3/benchmarks/ud_benchmark) on my own set of Akkadian sentences and I have encountered a difficulty when executing the 'assemble' command in the project.yml file. 

Following that model's requirements.txt file, I installed spacy-experimental 0.2.0 and spacy-transformers 1.1.3 in a virtual environment and managed to train the transformer and tokenizer (I did not do the 'extract' or 'convert' commands because I have my own data). However when I run the 'assemble command' I am told that spacy cannot find spacy-experimental.char_pretokenizer.v1:

```
RegistryError: [E893] Could not find function 'spacy-experimental.char_pretokenizer.v1' in function registry 'tokenizers'. If you're using a custom function, make sure the code is available. If the function is provided by a third-party package, e.g. spacy-transformers, make sure the package is installed in your environment.

Available names: spacy.Tokenizer.v1
```

The documentation in the readme file does not discuss any such issue. Does anyone know what might be the issue?

**Comment by adrianeboyd** [comment, marked as answer]
Can you double-check that `spacy-experimental` is installed in the environment you're running `spacy assemble` in? In the background the registry should work the exact same way as in `spacy train`.

**Reply by megamattc** [reply]
Hello,

I believe I do have spacy-experimental installed in my virtual environment, as the following command shows:

```
(udbench) (base) matthewong@Matthews-MacBook-Air ud_benchmark % pip show spacy-experimental
Name: spacy-experimental
Version: 0.2.0
Summary: Cutting-edge experimental spaCy components and features
Home-page: https://github.com/explosion/spacy-experimental
Author: Explosion
Author-email: contact@explosion.ai
License: MIT
Location: /Users/matthewong/Documents/nlp_stuff/spacy_projects/benchmarks/ud_benchmark/udbench/lib/python3.10/site-packages
Requires: spacy
Required-by: 
(udbench) (base) matthewong@Matthews-MacBook-Air ud_benchmark % 
```

**Reply by adrianeboyd** [reply]
At this point, our best (albeit frustrating) advice is to try installing all the required packages again in a new venv and see if that solves the problem.

**Reply by megamattc** [reply]
Hello again Adriane,

FYI: I did try reinstalling the required packages in a clean virtual environment on my local machine and it did not solve the problem. But when I tried it in Google Colab, first installing ```spacy-experimental``` and then ```spacy-transformers```, the ```assemble``` command did work properly. I don't know why, save for the fact that Google Colab has all sorts of other pre-installed packages I don't. I did not see other messages on this board talking about this problem, so I wonder if anyone else has actually tried to use the ud_benchmark package and found a problem...

**Reply by adrianeboyd** [reply]
Yeah, I think the preinstalled packages on colab make it trickier than if you were starting in a new venv. I also don't think that very many users have tested the `ud_benchmark` project, so there could be other bugs, but missing registered functions is usually just an install problem.

**Reply by megamattc** [reply]
Actually, in this case it appears it was the extra packages Colab provides that supplied the missing ```spacy-experimental.char_pretokenizer.v1```, compared to my local machine.

**Comment by megamattc** [comment]
Ok. Thank you for trying. I will try reinstalling it again.
